src/K4os.Compression.LZ4.Buffers/LZ4FrameDecoder.cs

```diff
@@ -129,8 +129,9 @@ public async ValueTask<int> DecodeAsync(PipeReader compressed, PipeWriter decomp
                 // Consume the block
                 totalConsumed += blockConsumed;
                 compressed.AdvanceTo(result.Buffer.GetPosition(offset: blockConsumed));
+                result = default;
```

Comment:
I got in the habit of clearing the `ReadResult` (and avoiding copying the reference to the buffer anywhere else) as soon as I call `AdvanceTo` because otherwise I'd tend to have race condition bugs when I'd accidentally dereference it later, when the memory backing the `ReadResult` got recycled on another thread.